apps/sms/views/new_message/js/new_message.js

```diff
@@ -0,0 +1,3031 @@
+      var threadPromise = Promise.reject();
+      if (parameters && parameters.number) {
+        threadPromise = MessageManager.findThreadFromNumber(parameters.number);
+      }
+
+      // The rejected promise will be returned in case we can't find thread
+      // for the specified number.
+      return threadPromise.then((id) => {
+        return Navigation.toPanel('thread', { id: id, focusComposer: true });
+      }, () => {
+        return Navigation.toPanel('composer', { activity: parameters });
+      });
+    });
+  },
+
+  _onNavigatingBack: function() {
+    this.stopRendering();
+
+    // We're waiting for the keyboard to disappear before animating back
+    return this._ensureKeyboardIsHidden().then(function() {
+      // Need to assimilate recipients in order to check if any entered
+      this.assimilateRecipients();
+
+      // If we're leaving a thread's message view,
+      // ensure that the thread object's unreadCount
+      // value is current (set = 0)
+      if (Threads.active) {
+        Threads.active.unreadCount = 0;
+      }
+
+      // If the composer is empty and we are either
+      // in an active thread or there are no recipients
+      // do not prompt to save a draft and remove saved drafts
+      // as the user deleted them manually
+      if (Compose.isEmpty() &&
+        (Threads.active || this.recipients.length === 0)) {
+        this.discardDraft();
+        return;
+      }
+
+      // If there is a draft and the content and recipients
+      // never got edited, re-save if threadless,
+      // then leave without prompting to replace
+      if (this.draft && !this.draft.isEdited) {
+        // Thread-less drafts are orphaned at this point
+        // so they need to be resaved for persistence
+        // Otherwise, clear the draft directly before leaving
+        if (!Threads.currentId) {
+          this.saveDraft();
+        } else {
+          this.draft = null;
+        }
+        return;
+      }
+
+      return this._showMessageSaveOrDiscardPrompt();
+    }.bind(this));
+  },
+
+  isKeyboardDisplayed: function conv_isKeyboardDisplayed() {
+    /* XXX: Detect if the keyboard is visible. The keyboard minimal height is
+     * 150px; when in reduced attention screen mode however the difference
+     * between window height and the screen height will be larger than 150px
+     * thus correctly yielding false here. */
+    return ((window.screen.height - window.innerHeight) > 150);
+  },
+
+  _ensureKeyboardIsHidden: function() {
+    if (this.isKeyboardDisplayed()) {
+      return new Promise(function(resolve) {
+        var setTimer = window.setTimeout(resolve, 400);
+        window.addEventListener('resize', function keyboardHidden() {
+          window.clearTimeout(setTimer);
+          window.removeEventListener('resize', keyboardHidden);
+          resolve();
+        });
+      });
+    }
+    return Promise.resolve();
+  },
+
+  _showMessageSaveOrDiscardPrompt: function() {
```

Comment:
```_showMessageSaveOrDiscardPrompt``` is going to be removed, so I don't really care if generalize it somehow or just duplicate in both views for now.